pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+	dimensions := map[string]string{
+		"operationtype":     operationType.String(),
+		"provisioningstate": provisioningState.String(),
+		"resulttype":        string(ocb.getResultType(backendErr)),
+	}
+
+	ocb.gatherCorrelationID(doc, dimensions)
+	ocb.gatherMiscMetrics(doc, dimensions)
+	ocb.gatherAuthMetrics(doc, dimensions)
+	ocb.gatherNetworkMetrics(doc, dimensions)
+	ocb.gatherNodeMetrics(doc, dimensions)
+
+	ocb.logMetricDimensions(log, dimensions)
+	ocb.m.EmitGauge(metricInstall, 1, dimensions)
+}
+
+func (ocb *openShiftClusterBackend) getResultType(backendErr error) utillog.ResultType {
+	var resultType utillog.ResultType
+	err, ok := backendErr.(*api.CloudError)
+	if ok {
+		resultType = utillog.MapStatusCodeToResultType(err.StatusCode)
+	}
+	return resultType
+}
+
+func (ocb *openShiftClusterBackend) logMetricDimensions(log *logrus.Entry, dimensions map[string]string) {
+	for metric, value := range dimensions {
+		log.Info(fmt.Sprintf("%s: %s = %s", metricPackage, metric, value))
+	}
+}
+
+func (m *openShiftClusterBackend) gatherCorrelationID(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.CorrelationData != nil {
+		dimensions["correlationid"] = doc.CorrelationData.CorrelationID
+		dimensions["client_requestid"] = doc.CorrelationData.ClientRequestID
+		dimensions["requestid"] = doc.CorrelationData.RequestID
+		dimensions["async_operationsid"] = doc.AsyncOperationID
+	} else {
+		dimensions["correlationid"] = empty
+		dimensions["client_requestid"] = empty
+		dimensions["async_operationsid"] = empty
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherMiscMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	dimensions["subscriptionid"] = ocb.env.SubscriptionID()
+	dimensions["resourceid"] = doc.ResourceID
+	dimensions["location"] = doc.OpenShiftCluster.Location
+	dimensions["resourcegroup"] = doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+	dimensions["name"] = doc.OpenShiftCluster.Name
+	dimensions["ocpversion"] = doc.OpenShiftCluster.Properties.ClusterProfile.Version
+	dimensions["workercount"] = strconv.FormatInt(int64(len(doc.OpenShiftCluster.Properties.WorkerProfiles)), 10)
+	dimensions["rpversion"] = doc.OpenShiftCluster.Properties.ProvisionedBy
+
+	for flag, feature := range doc.OpenShiftCluster.Properties.OperatorFlags {
+		dimensions[fmt.Sprintf("operator-%s", flag)] = feature
+	}
+
+	if doc.OpenShiftCluster.Tags != nil {
+		dimensions["tags"] = enabled
+	} else {
+		dimensions["tags"] = disabled
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherNodeMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.MasterProfile.DiskEncryptionSetID != "" {
+		dimensions["masterprofile.diskencryptionsetid"] = enabled
+	} else {
+		dimensions["masterprofile.diskencryptionsetid"] = disabled
+	}
```

Comment:
IMO we don't need a dimension for this - we can just utilize whether or not encryptionathost is enabled for this purpose.

Reply:
This is a common Cluster Install Failure, it would be good to capture the rate at which it happens

Reply:
My understanding might have been incorrect - looking through our code it looks like it is possible to either set or omit diskencryptionsetid when encryption at host is enabled.